File: 010-Summation-of-primes.c
#include <stdio.h>

int main(int argc, const char *argv[])
{
   unsigned int   list[2000000]={0}, ii, jj;

   unsigned int   sum = 0;

   for(ii = 2; ii < 2000000; ii++){
      if(list[ii] == 0){ /* If prime */
         sum += ii;
         for(jj = ii; jj < 2000000; jj+=ii) /* Kill all multiples */
            list[jj] = 1;
      }
   }

   printf("% d % d\n", ii, sum);
   return 0;
}

File: 010-SummationOfPrimes.py
#! /usr/bin/env python

import numpy as np

def seiveOfEratosthenes():
    """
    Implements the Seive of Eratosthenes to generate a list of first 2 million
    prime numbers
    """

    n = 2000000
    x = list(range(2, n))

    for nn in range(0, n-2):
        if x[nn] == 0: # Not a prime, don't bother with factors
            continue

        step = x[nn]
        for xx in range(nn+step, n-2, step):
            x[xx] = 0 # Set all multiples to zero

    sump = sum(x)
    return sump

sump = seiveOfEratosthenes()
print (sump)
